src/odemis/driver/test/actuator_test.py

```diff
@@ -548,6 +548,45 @@ def test_metadata(self):
 
         testing.assert_pos_almost_equal(stage.speed.value, {"x": 10e-6, "y": 20e-6})
 
+    def test_get_axes_range(self):
+        """
+        Test the range calculation in ConvertStage
+        """
+        dependency = simulated.Stage("stage", "test", axes=["a", "b"],
+                                     ranges={"a": [0.e-3, 100.e-3], "b": [0.e-3, 13.e-3]})
+        # Map dependency onto a given rotation
+        tilt = math.radians(26)
+        stage = ConvertStage("conv", "align", {"orig": dependency},
+                             axes=["a", "b"], rotation=tilt)
+        # Move stage within the range of dependency
+        first_value = 2e-03
+        second_value = 0
+        f = stage.moveAbs({"x": first_value, "y": second_value})
+        f.result()
+
+        # Tests
+        # Move convert stage outside the range of its dependency to raise exception
+        with self.assertRaises(ValueError):
+            f = stage.moveAbs({"x": -5.e-03, "y": second_value})
+            f.result()
+        # Check if the stage is at desired location
+        testing.assert_pos_almost_equal(stage.position.value, {"x": first_value, "y": second_value})
+        new_first_value = first_value * math.cos(tilt)
+        new_second_value = first_value * math.sin(tilt)
+        testing.assert_pos_almost_equal(dependency.position.value, {"a": new_first_value, "b": new_second_value})
+        # Move the stage in FM imaging range at different points
+        # and check if ConvertStage coordinates fall in between the ConvertStage range
+        f = dependency.moveAbs({"a": 30.e-3, "b": 0.e-3})
+        f.result()
+        # Move outside the given range of b, an error from ConvertStage must raise
+        with self.assertRaises(ValueError):
+            f = dependency.moveAbs({"a": 30.e-3, "b": 15.e-3})
+            f.result()
+
+        # Reset the position of dependency
+        f = stage.moveAbs({"x": first_value, "y": second_value})
+        f.result()
```

Comment:
Why do you do that? What is it testing? Typically, it's always going to work, as you've just tried it a few lines before.

Reply:
This is no longer required. I will remove it. I used it when I was running the test repeatedly and using moveRel instead of moveAbs. In that case, the absolute stage position was changing, making it difficult to debug.